pkg/kv/kvserver/client_raft_test.go

```diff
@@ -5396,6 +5396,11 @@ func TestProcessSplitAfterRightHandSideHasBeenRemoved(t *testing.T) {
 							// Newly-started stores (including the "rogue" one) should not GC
 							// their replicas. We'll turn this back on when needed.
 							DisableReplicaGCQueue: true,
+							// Some subtests, e.g. (4), expect that n1 catches up on the raft
+							// log after a partition, and runs the split trigger. Disable raft
+							// log truncation to make sure that it doesn't miss the split
+							// trigger by being caught up by a later snapshot. See #154313.
+							DisableRaftLogQueue:   true,
```

Comment:
Only thing I'm not sure is whether the truncation actually happened in this flake. Our truncation policy is quite [generous](https://github.com/cockroachdb/cockroach/blob/0d5068e114763c15f4d67bbdfe20a85b654ad2c8/pkg/kv/kvserver/raft_log_queue.go#L554-L558), it needs 64KiB of backlog before truncating. In this test, there is very little written to this range.

But it's the only plausible explanation I see. Maybe there is some rare condition that made the truncation happen.